ako-gateway-api/nodes/gateway_model_rel.go

```diff
@@ -238,7 +244,10 @@ func HTTPRouteToGateway(namespace, name, key string) ([]string, bool) {
 	var gatewayList []string
 	var hostnameIntersection []string
 	var gwNsNameList []string
-	for _, parentRef := range hrObj.Spec.ParentRefs {
+	for index, parentRef := range hrObj.Spec.ParentRefs {
+		if hrObj.Status.Parents[index].Conditions[0].Type == string(gatewayv1.GatewayConditionAccepted) && hrObj.Status.Parents[index].Conditions[0].Status == metav1.ConditionFalse {
```

Comment:
is it guaranteed that this condition here is same as the status condition on gateway? will it be better to fetch the gateway status condition here again?

Reply:
Its' a good catch. I have mostly copied same conditions every. It should rather be gatewayv1.RouteConditionAccepted which is guaranteed to be prrsent always since validation layer sets it. Have made changes

Reply:
if we are enqueueing the status then we can't guarantee that the status will get updated before the key is processed in graph layer.